client/packages/programs/src/JsonForms/useJsonForms.tsx

```diff
@@ -138,9 +143,13 @@ export const useJsonForms = <R,>(
 
     // Run mutation...
     try {
+      await formActions.run({ preSubmit: true });
+
       const sanitizedData = stripEmptyAdditions(initialData, data);
       const result = await save?.(sanitizedData);
 
+      await formActions.run({ preSubmit: false });
+
```

Comment:
As suggested earlier, we now have the option to specify actions as "pre-submit" or "post-submit".

In future we might want to do some more error handling, and abort the save if any (or all?) of the pre-submit actions fail. But we need more real world use cases before that becomes worthwhile.